File: app/src/main/java/br/edu/farol/gadoplus/adapter/PropriedadeAdapter.java
package br.edu.farol.gadoplus.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.recyclerview.widget.RecyclerView;
import java.util.List;
import br.edu.farol.gadoplus.R;
import br.edu.farol.gadoplus.model.Propriedade;

public class PropriedadeAdapter extends RecyclerView.Adapter<PropriedadeAdapter.PropriedadeViewHolder> {

    private List<Propriedade> propriedades;
    private OnItemClickListener listener;

    @Override
    public PropriedadeViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.recyclerview_item, parent, false);
        return new PropriedadeViewHolder(itemView);
    }


    @Override
    public void onBindViewHolder(PropriedadeViewHolder viewHolder, int i) {

        viewHolder.nome.setText(propriedades.get(i).getNome());
        viewHolder.descricao.setText(propriedades.get(i).getDescricao());
        viewHolder.hectares.setText(propriedades.get(i).getHectares() + " hec");

    }

    @Override
    public int getItemCount() {
        if (propriedades != null)
            return propriedades.size();
        else return 0;
    }

    public void setPropriedade(List<Propriedade> propriedades) {
        this.propriedades = propriedades;
        notifyDataSetChanged();
    }

    class PropriedadeViewHolder extends RecyclerView.ViewHolder {

        private TextView nome, descricao, hectares;

        public PropriedadeViewHolder(View itemView) {
            super(itemView);
            nome = itemView.findViewById(R.id.text_view_title);
            descricao = itemView.findViewById(R.id.text_view_description);
            hectares = itemView.findViewById(R.id.text_view_priority);

            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    int position = getAdapterPosition();
                    if (listener != null && position != RecyclerView.NO_POSITION)
                        listener.onItemClick(propriedades.get(position));
                }
            });
        }
    }

    public interface OnItemClickListener {
        void onItemClick(Propriedade propriedade);
    }

    public void setOnClickListener(OnItemClickListener listener) {
        this.listener = listener;
    }

}

File: app/src/main/java/br/edu/farol/gadoplus/adapter/PropriedadeAdapter.kt
package br.edu.farol.gadoplus.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import br.edu.farol.gadoplus.R
import br.edu.farol.gadoplus.adapter.PropriedadeAdapter.PropriedadeViewHolder
import br.edu.farol.gadoplus.model.*

class PropriedadeAdapter : RecyclerView.Adapter<PropriedadeViewHolder>() {
    private var propriedades: List<Propriedade>? = null
    private var listener: OnItemClickListener? = null
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PropriedadeViewHolder {
        val itemView = LayoutInflater.from(parent.context)
                .inflate(R.layout.recyclerview_item, parent, false)
        return PropriedadeViewHolder(itemView)
    }

    override fun onBindViewHolder(viewHolder: PropriedadeViewHolder, i: Int) {
        viewHolder.nome.text = propriedades!![i].nome
        viewHolder.descricao.text = propriedades!![i].descricao
        viewHolder.hectares.setText("${propriedades!![i].hectares}  hec")
    }

    override fun getItemCount(): Int {
        return if (propriedades != null) propriedades!!.size else 0
    }

    fun setPropriedade(propriedades: List<Propriedade>?) {
        this.propriedades = propriedades
        notifyDataSetChanged()
    }

    inner class PropriedadeViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val nome: TextView
        val descricao: TextView
        val hectares: TextView

        init {
            nome = itemView.findViewById(R.id.text_view_title)
            descricao = itemView.findViewById(R.id.text_view_description)
            hectares = itemView.findViewById(R.id.text_view_priority)
            itemView.setOnClickListener {
                val position = adapterPosition
                if (listener != null && position != RecyclerView.NO_POSITION) listener!!.onItemClick(propriedades!![position])
            }
        }
    }

    interface OnItemClickListener {
        fun onItemClick(propriedade: Propriedade?)
    }

    fun setOnClickListener(listener: OnItemClickListener?) {
        this.listener = listener
    }
}